doc: release: add remaining release notes for TF-M, and list of new boards

ioannisg opened the pull request:
Add remaining release notes entries for TF-M for Zephyr
v2.6.0 release, mentioning the added samples and platforms.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

galak commented on doc/releases/release-notes-2.6.rst:
```diff
@@ -376,6 +376,12 @@ Trusted Firmware-m
   (Cortex-M33 Non-Secure) with TF-M as the secure firmware component.
 * Synchronized Trusted-Firmware-M module to the upstream v1.3.0 release.
+* Added a new sample to run the PSA tests with Zephyr
+* Added a new sample to run the TF-m regression tests using the Zephyr build system
```
Nit, `TF-m` -> `TF-M`

ioannisg replied:
fixed now

ioannisg committed:
doc: release notes: add Zephyr v2.6.0 release notes for TF-M

Add release note entries for TF-M.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
doc: release: add note for new TF-M boards in v.2.6.0

Add a release note entry for the new Zephyr boards
supported in TF-M in Zephyr v.2.6.0.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

erwango reviewed (approved):
Reference to https://github.com/zephyrproject-rtos/zephyr/pull/34868 could be added once merged.

ioannisg commented:
> Reference to #34868 could be added once merged.

Sure, and not only that - we might have to add some info about the mbedtls version we will be using in tfm. 
@erwango ok if we get this in for now, and revisit the release notes before the 2.6 release?
